src/configuration/testcafe-configuration.ts

```diff
@@ -34,7 +34,12 @@ import WarningLog from '../notifications/warning-log';
 import browserProviderPool from '../browser/provider/pool';
 import BrowserConnection, { BrowserInfo } from '../browser/connection';
 
-const CONFIGURATION_FILENAME = '.testcaferc.json';
+const BASE_CONFIGURATION_FILENAME       = '.testcaferc';
+const PRIORITY_CONFIGURATION_EXTENSIONS = {
+    js:   0,
```

Comment:
Also, I propose to extract the `js` extension to constant and reuse it [here] (https://github.com/DevExpress/testcafe/pull/6319/files#diff-829eee50f2be6bc3b9b60ea5ceec1cd3dd099bb6c47010e5d6a2f3355dfed9cdR184).

```js
const BASE_CONFIGURATION_FILENAME  = `.testcaferc`;
const JS_CONFIGURATION_EXTENTION   = '.js'; 
const JSON_CONFIGURATION_EXTENTION = '.json';
const CONFIGURATION_EXTENTIONS     = [JS_CONFIGURATION_EXTENTION, JSON_CONFIGURATION_EXTENTION];
const CONFIGURATION_FILENAMES      = CONFIGURATION_EXTENTIONS.map(ext => BASE_CONFIGURATION_FILENAME+ ext);
```